File: app/src/main/java/com/example/yc/zhihudaily/ConnectingTheInternet/GetApi.java
package com.example.yc.zhihudaily.ConnectingTheInternet;

import com.example.yc.zhihudaily.Model.Datas;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface GetApi {
    @GET("stories/latest")
    Call<Datas> getList();

    @GET("news/before/{date}")
    Call<Datas> getList(@Path("date") String date);
}

File: app/src/main/java/com/example/yc/zhihudaily/ConnectingTheInternet/GetApi.kt
package com.example.yc.zhihudaily.ConnectingTheInternet

import com.example.yc.zhihudaily.Model.Datas

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface GetApi {
    @get:GET("stories/latest")
    val list: Call<Datas>

    @GET("news/before/{date}")
    fun getList(@Path("date") date: String): Call<Datas>
}
